### PR TITLE
Bump replicaset to apps/v1

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**:
This PR bumps replicaset to apps/v1.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: https://github.com/kubernetes/features/issues/484

**Release note**:

```release-note
Bump ReplicaSet to apps/v1.
```

### CODE REVIEW

**Comment by crimsonfaith91** [comment]
/cc @janetkuo 
All development codes are at commit #6: `bump replicasets to apps/v1`
Thanks!

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 1]
```diff
@@ -205,3 +205,130 @@ type DaemonSetList struct {
 	// A list of daemon sets.
 	Items []DaemonSet `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
```
Comment (kow3ns) [inline staging/src/k8s.io/api/apps/v1/types.go]: This is not generated. It should be in your initial commit.

Comment (crimsonfaith91) [reply staging/src/k8s.io/api/apps/v1/types.go]: you are right. Same for `staging/src/k8s.io/api/apps/v1/register.go` too.

**Comment by janetkuo** [comment]
@crimsonfaith91 please investigate test failures

**Comment by crimsonfaith91** [comment]
@janetkuo Sure! :)

**Comment by crimsonfaith91** [comment]
There is a leader election issue resulting in many timeouts. Will have a look at it tomorrow once the DS PR merges.

**Comment by kow3ns** [comment]
@crimsonfaith91 did you for this from the feature branch originally?

**Comment by crimsonfaith91** [comment]
Yes! For statefulset too! I was told that either one is fine, so I sticked with feature branch.

**Comment by kow3ns** [comment]
#53278 has a verify issue similar to the one you have. It will have to be manually updated or rebased against master to pass verify due to #53481.

**Comment by crimsonfaith91** [comment]
@kow3ns Yup! Thanks!

**Comment by crimsonfaith91** [comment]
@kow3ns Despite forking from the feature repo, this PR is rebased against and target to master branch.

**Comment by crimsonfaith91** [comment]
Most failures are due to a leader election lost issue. Pasting the error here for reference:
```
E1005 16:43:39.934552   69447 leaderelection.go:224] error retrieving resource lock kube-system/kube-controller-manager: Get http://127.0.0.1:8080/api/v1/namespaces/kube-system/endpoints/kube-controller-manager: dial tcp 127.0.0.1:8080: getsockopt: connection refused
E1005 16:43:39.934590   69447 event.go:260] Could not construct reference to: '&v1.Endpoints{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Subsets:[]v1.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' 'LeaderElection' 'jt-linux.svl.corp.google.com stopped leading'
I1005 16:43:39.934687   69447 leaderelection.go:203] failed to renew lease kube-system/kube-controller-manager: timed out waiting for the condition
F1005 16:43:39.934747   69447 controllermanager.go:194] leaderelection lost
```

**Comment by crimsonfaith91** [comment]
It is weird. When I ran the `hack/make-rules/test-cmd.sh` tests individually, the leader election is not lost. However, when I ran all the tests in one go, it is lost.

**Comment by crimsonfaith91** [comment]
For reference, the leader election problem is due to not adding this line at `strategy.go`: https://github.com/kow3ns/kubernetes/blob/47bb984f4848d6624d60a895058be3d2fe95d115/pkg/registry/extensions/replicaset/strategy.go#L131

I referred daemonset codes (first time bumping a controller to a new version), but the line was not added to daemonset at that moment, so missed it.

**Comment by crimsonfaith91** [comment]
closing in dup of https://github.com/kubernetes/kubernetes/pull/53679/

**Pull request closed without merging**